modules/gltf/structures/gltf_node.cpp

```diff
@@ -203,17 +203,36 @@ NodePath GLTFNode::get_scene_node_path(Ref<GLTFState> p_state, bool p_handle_ske
 	Vector<StringName> subpath;
 	Ref<GLTFNode> current_gltf_node = this;
 	const int gltf_node_count = p_state->nodes.size();
-	if (p_handle_skeletons && skeleton != -1) {
-		// Special case for skeleton nodes, skip all bones so that the path is to the Skeleton3D node.
-		// A path that would otherwise be `A/B/C/Bone1/Bone2/Bone3` becomes `A/B/C/Skeleton3D:Bone3`.
-		subpath.append(get_name());
-		// The generated Skeleton3D node will be named Skeleton3D, so add it to the path.
-		path.append("Skeleton3D");
-		do {
-			const int parent_index = current_gltf_node->get_parent();
-			ERR_FAIL_INDEX_V(parent_index, gltf_node_count, NodePath());
-			current_gltf_node = p_state->nodes[parent_index];
-		} while (current_gltf_node->skeleton != -1);
+	if (p_handle_skeletons) {
+		if (skin != -1) {
+			// Special case for skinned meshes, they become a child of a Skeleton3D node.
```

Comment:
This code is quite ugly.

Reply:
What do you want me to do about it?

Reply:
I think it has to be fixed to run before the skin tool.